cromwell_tools/cromwell_tools.py

```diff
@@ -208,3 +219,58 @@ def read_local_file(path):
     with open(path) as f:
         contents = f.read()
     return contents
+
+
+def validate_cromwell_label(label_object):
+    """Check if the label object is valid for Cromwell.
+
+    Note: this function as well as the global variables _CROMWELL_LABEL_LENGTH, _CROMWELL_LABEL_KEY_REGEX
+        and _CROMWELL_LABEL_VALUE_REGEX are implemented based on the Cromwell's documentation:
+        https://cromwell.readthedocs.io/en/develop/Labels/ and the Cromwell's code base:
+        https://github.com/broadinstitute/cromwell/blob/master/core/src/main/scala/cromwell/core/labels/Label.scala#L16
+        Both the docs and the code base of Cromwell could possibly change in the future, please update this
+        checker on demand.
+    :param label_object: A dictionary or a key-value object string that define a Cromwell label.
+    """
+    err_msg = ""
+
+    if isinstance(label_object, str) or isinstance(label_object, bytes):
+        label_object = json.loads(label_object)
+    elif isinstance(label_object, io.BytesIO):
+        label_object = json.loads(label_object.getvalue())
+
+    for label_key, label_value in label_object.items():
+        err_msg += _label_content_checker(_CROMWELL_LABEL_KEY_REGEX, label_key)
+        err_msg += _label_content_checker(_CROMWELL_LABEL_VALUE_REGEX, label_value)
+        err_msg += _label_length_checker(_CROMWELL_LABEL_LENGTH, label_key)
+        err_msg += _label_length_checker(_CROMWELL_LABEL_LENGTH, label_value)
+
+    if err_msg == "":
+        return True
+    else:
+        raise ValueError(err_msg)
+
+
+def _label_content_checker(regex, content):
```

Comment:
I would put these three functions above `validate_cromwell_label` since they are called by that function. I would also add docstrings with parameter typing information.